balboa-client-kafka/src/main/java/com/socrata/metrics/impl/MetricLoggerJava.java

```diff
@@ -0,0 +1,62 @@
+package com.socrata.metrics.impl;
+
+import com.blist.metrics.impl.queue.AbstractJavaMetricQueue;
+import com.socrata.balboa.metrics.Metric;
+import com.socrata.balboa.metrics.config.Configuration;
+import com.socrata.metrics.IdParts;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Java wrapper around MetricLoggerToKafka.
+ */
+public class MetricLoggerJava extends AbstractJavaMetricQueue {
+
+    /**
+     * Mapping topics => MetricLoggers.  Allows for the
+     */
+    private static Map<String, MetricLoggerJava> instances = new HashMap<>();
+
+    /**
+     * Underlying Metrics logger
+     */
+    private MetricLoggerToKafka.MetricLogger logger;
+
+    /**
+     * Returns a Metric Logger Instance for a given topic.
+     *
+     * @param topic Topic to get Metric Logger for.
+     * @return The producer for a given topic.
+     */
+    public static MetricLoggerJava getInstance(String topic) {
+        if (instances.containsKey(topic)) {
+            return instances.get(topic);
+        } else {
+            MetricLoggerJava logger = new MetricLoggerJava(topic);
```

Comment:
this should be in a synchronized block; with a check just before the creation to see if the topic was already added